### ISSUE TITLE
SegFault cloning Space Python handle, only with release build

### ISSUE THREAD

**Issue by luketpeterson** [trueagi-io/hyperon-experimental#405]
This problem has existed for quite a while, but it only occurs with a release build.

Steps to reproduce:
```
cd hyperon-experimental
rm -r build
mkdir build
cmake -DCMAKE_BUILD_TYPE=Release ../ #as opposed to just `cmake ../`
make
make check
```

The `test_match_nested_grounding_space` and `test_match_nested_custom_space` Python tests both crash when trying to clone the Space handle.  I have validated that the same operations performed by the C API are fine.  Test implemented in https://github.com/trueagi-io/hyperon-experimental/pull/404

I am currently debugging this.  My best guess is that, somehow, the space gets freed (refcount reaches 0) prematurely, but this hypothesis is unconfirmed.

**Comment by luketpeterson** [comment]
Actually it was just a dumb bug.  Should be fixed in 1681a3e

**Comment by luketpeterson** [comment]
Resolved and merged
